File: Neetcode 150/Intervals/non-overlapping-intervals.cpp

// https://leetcode.com/problems/non-overlapping-intervals/description/

bool comp(vector<int>&a, vector<int>&b){
    return a[1] < b[1];
}

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        
        int ans = -1;
        int size = intervals.size();
        if(size == 0) return ans;

        // sorting accoring to second element in array
        sort(intervals.begin(), intervals.end(), comp);

        auto prev = intervals[0];
        for(auto ele : intervals){
            if(prev[1] > ele[0]) ans++;
            else prev = ele;
        }
        
        return ans;
    }
};

File: Neetcode-150-js/Intervals/non-overlapping-intervals.js

// https://leetcode.com/problems/non-overlapping-intervals/

/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    let count = 0;
    let len = intervals.length;
    if(len === 0) return count;

    intervals = intervals.sort((a, b) => a[1] - b[1]);
    let prev = intervals[0];
    for(let i = 1; i < len; i++){
        if(prev[1] > intervals[i][0]) count++;
        else prev = intervals[i];
    }

    return count;
};